src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -543,44 +576,55 @@ public Task RelayTransaction(uint256 hash)
 
             // before locking an exclusive task
             // check if there is anything to processes
-            if (!await this.manager.MempoolLock.ReadAsync(() => this.inventoryTxToSend.Keys.Any()))
-            {
-                this.logger.LogTrace("(-)[NO_TXS]");
-                return;
+            using (this.asyncLock.Lock())
```

Comment:
Please delete 2 comments above that line. 
Merge this lock block with the one after that.